src/validation_workflow/docker/validation_docker.py

```diff
@@ -0,0 +1,202 @@
+        # replace the placeholder in template docker-compose.yaml which is taken from sample in https://opensearch.org/docs/latest/opensearch/install/docker/#sample-docker-composeyml
+        self.yml_template = "docker-compose.yml"
+        self.yml_template_location = r"./src/validation_workflow/docker/"
+        self.yml_file = os.path.join(self.yml_template_location, self.yml_template)
+
+        self.tmp_dir = TemporaryDirectory()
+        shutil.copy2(self.yml_file, self.tmp_dir.name)
+        target_yml_file = self.tmp_dir.name + '/' + self.yml_template
+        self.inplace_change(target_yml_file, 'OS_IMAGE_PLACEHOLDER', OpenSearch_image + ":" + version)
+        self.inplace_change(target_yml_file, 'OSD_IMAGE_PLACEHOLDER', OpenSearchDashboard_image + ":" + version)
+        self.inplace_change(target_yml_file, 'OS1_NAME', OpenSearchNode1_name)
+        self.inplace_change(target_yml_file, 'OS2_NAME', OpenSearchNode2_name)
+        self.inplace_change(target_yml_file, 'OSD_NAME_PLACEHOLDER', OpenSearchDashboard_name)
+
+        # spin up containers
+        docker_compose_up = 'docker-compose -f ' + target_yml_file + ' up -d'
+        result = subprocess.run(docker_compose_up, shell=True, stdout=PIPE, stderr=PIPE, universal_newlines=True)
+        return ('returncode=0' in (str(result)), target_yml_file)
+
+    def inplace_change(self, filename: str, old_string: str, new_string: str) -> None:
+
+        with open(filename) as f:
+            s = f.read()
+            if old_string not in s:
+                print('"{old_string}" not found in {filename}.'.format(**locals()))
+                return
+
+        with open(filename, 'w') as f:
+            print('Changing "{old_string}" to "{new_string}" in {filename}'.format(**locals()))
+            s = s.replace(old_string, new_string)
+            f.write(s)
+
+    def get_image_id(self, image_name: str, image_version: str) -> str:
+        local_inspect = "docker image inspect -f '{{ .Id }}' " + image_name + ":" + image_version
+
+        # Removing the image if it exists at local
+        result_inspect = subprocess.run(local_inspect, shell=True, stdout=PIPE, stderr=PIPE, universal_newlines=True)
+        if (result_inspect.returncode == 0):
+            logging.info('Image exists at local : ' + result_inspect.stdout + ' : ' + image_name + ":" + image_version)
+            logging.info(f'Overwriting the local image by pulling a fresh {image_name}:{image_version} from {self.args.docker_source}')
+        else:
+            logging.info(f'Image does not exist at local, proceed with Pull from the {self.args.docker_source}')
+
+        return (self.pull_image(image_name, image_version))
+
+    def pull_image(self, image_name: str, image_version: str) -> str:
+
+        dockerHub_pull = "docker pull " + image_name + ":" + image_version
+        local_inspect = "docker image inspect -f '{{ .Id }}' " + image_name + ":" + image_version
```

Comment:
Why not use `f` string but trying to `+` everything?

Reply:
This applied to any similar issues in this PR.

Reply:
Thanks, @peterzhuamazon. Let us use f-strings in those cases..

Reply:
Thanks, @peterzhuamazon. Let us use f-strings in those cases..